File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.java
import java.io.*;
import java.util.*;

class Solution {
    public int solution(int k, int[] tangerine) {
        int answer = 0;
        Map<Integer, Integer> map = new HashMap<>();
        
        //크기에 따른 귤 개수 카운트하여 저장
        for(int i=0; i<tangerine.length; i++){
            map.put(tangerine[i], map.getOrDefault(tangerine[i],0)+1);
        }
        
        //Map.Entry를 리스트로 변환
        List<Map.Entry<Integer, Integer>> entryList = new ArrayList<>(map.entrySet());
        
        //value 값에 따라 내림차순으로 정렬
        entryList.sort((e1, e2) -> e2.getValue().compareTo(e1.getValue()));
        
        //정렬된 결과에 따라 귤 고르는 로직
        for(Map.Entry<Integer, Integer> entry : entryList){
            if(k<=0) break;
            k -= entry.getValue();
            answer++;
        }
        
        return answer;
    }
}

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.py
def solution(k, tangerine):
    answer = 0
    
    #1. 귤 크기 별 딕셔너리 구축
    t_dict = {}
    
    for t in tangerine:
        t_dict[t] = t_dict.get(t, 0) + 1
    
    #2. 개수 많은 순으로 내림차순 정렬
    t_dict = sorted(t_dict.items(), key=lambda x: x[1], reverse = True)
    
    #3. 귤 k개 판매
    picked = 0
    for size, count in t_dict:
        picked += count
        answer +=1
        
        if picked >= k:
            break
    return answer


